pkg/controller/controller_instance.go

```diff
@@ -468,6 +514,10 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1beta1.ServiceIns
 	glog.V(4).Info(pcb.Message("Processing deleting event"))
 
 	instance = instance.DeepCopy()
+	// Any status updates from this point should have an updated observed generation
+	if instance.Status.ObservedGeneration != instance.Generation {
+		c.prepareObservedGeneration(instance)
```

Comment:
Yeah, I agree that we might just stop updating `ObservedGeneration` for both orphan mitigation and actual deletion. The latter is the one that is triggered by the user though, so it might be still better to report that we have started processing the deletion of instance.
Will add a condition on orphan mitigation there (ugly but should work).